File: api/src/main/java/net/funnyguilds/drama/discord/webhook/DiscordWebhookFactory.java
package net.funnyguilds.drama.discord.webhook;

import org.springframework.stereotype.Service;

import java.util.Collections;

@Service
public class DiscordWebhookFactory {

    public DiscordWebhook createDramaAlert(String drama) {
        DiscordWebhook webhook = new DiscordWebhook();
        DiscordWebhooksEmbed embed = new DiscordWebhooksEmbed();
        embed.setTitle("Latest Drama News");
        embed.setDescription(drama);
        webhook.setEmbeds(Collections.singletonList(embed));
        return webhook;
    }
}

File: api/src/main/java/net/funnyguilds/drama/discord/webhook/DiscordWebhookFactory.kt
package net.funnyguilds.drama.discord.webhook

import org.springframework.stereotype.Service

@Service
class DiscordWebhookFactory {

    fun createDramaAlert(drama: String): DiscordWebhook {
        return DiscordWebhook(null, listOf(DiscordWebhooksEmbed("Latest Drama News", drama)))
    }
}
